src/huggingface_hub/constants.py

```diff
@@ -129,3 +129,41 @@ def _as_int(value: Optional[str]) -> Optional[int]:
 HF_HUB_LOCAL_DIR_AUTO_SYMLINK_THRESHOLD: int = (
     _as_int(os.environ.get("HF_HUB_LOCAL_DIR_AUTO_SYMLINK_THRESHOLD")) or 5 * 1024 * 1024
 )
+
+# List frameworks that are handled by the InferenceAPI service. Useful to scan endpoints and check which models are
+# deployed and running. Since 95% of the models are using the top 4 frameworks listed below, we scan only those by
+# default. We still keep the full list of supported frameworks in case we want to scan all of them.
+MAIN_INFERENCE_API_FRAMEWORKS = [
```

Comment:
I think we have a confusion between what we consider a "framework" on the Hub and what is considered as "framework" under the hood for InferenceAPI. As I see it, `list_deployed_models` should be a helper to help discoverability of models already deployed so that the user don't have to wait for the model to load. I don't think we should choose the `MAIN_INFERENCE_API_FRAMEWORKS` list based on the Hub distribution but rather on the InferenceAPI distribution. I know it is only a proxy but if I check every possible inference framework, I get this:

```
diffusers 112
sentence-transformers 13
text-generation-inference 20
transformers 298
adapter-transformers 0
allennlp 0
asteroid 1
bertopic 0
doctr 0
espnet 2
fairseq 4
fastai 0
fasttext 0
flair 2
generic 2
k2 0
keras 0
mindspore 0
nemo 1
open_clip 0
paddlenlp 0
peft 0
pyannote-audio 1
sklearn 0
spacy 0
span-marker 0
speechbrain 3
stanza 0
timm 1
```

Which is why I chose those 4 frameworks. Also AFAIK, TGI models deployed on InferenceAPI is in fact a curated list of models that we consider as worthy to run which makes it even more important to list by default IMO (since users cannot spin-up LLMs by themselves the same way as any small-enough diffusers/transformers model)

Reply:
Discussed offline. The current solution is not ideal (might lead to confusion between Hub frameworks vs InferenceAPI frameworks) but hopefully it shouldn't be too bad as we (I) expect most users to not use the `frameworks` parameters and just take the default output. I added two `Tip` sections in the docstring to remind that `list_deployed_models` and `get_model_status` are meant to be complementary (the first one for discoverability, the second one for users that know what they want).